contracts/MintShop1155.sol

```diff
@@ -678,14 +679,14 @@ contract MintShop1155 is Sweepable, ReentrancyGuard {
   */
   function getWhitelistStatus(uint256[] calldata _ids,
     address[] calldata _addresses) external view returns (bool[][] memory) {
-    bool[][] memory whitelistStatus;
+    bool[][] memory whitelistStatus = new bool[][](_ids.length);
     for (uint256 i = 0; i < _ids.length; i++) {
+      whitelistStatus[i] = new bool[](_addresses.length);
       uint256 id = _ids[i];
       uint256 whitelistVersion = whitelists[id].currentWhitelistVersion;
       for (uint256 j = 0; j < _addresses.length; j++) {
-        bytes32 addressKey = keccak256(abi.encode(whitelistVersion,
-          _addresses[j]));
-        whitelistStatus[j][i] = whitelists[id].addresses[addressKey];
+        bytes32 addressKey = keccak256(abi.encode(whitelistVersion, _addresses[j]));
+        whitelistStatus[i][j] = whitelists[id].addresses[addressKey];
```

Comment:
Good catch with the mapping order being reversed. Our front-end needs are such that it's preferable to receive data structured where we can index with the address as the first field, but this reversal made it hard to correlate the output with the input. I'd like to keep the mapping its current address-first id-second ordering, but rearrange the ordering of the `_addresses` and `_ids` arrays to make that output make more sense.